test/tempfile.go

```diff
@@ -20,23 +20,24 @@ func (s *TempFileSuite) TestTempFile(c *C) {
 	f, err := s.FS.TempFile("", "bar")
 	c.Assert(err, IsNil)
 	c.Assert(f.Close(), IsNil)
-	c.Assert(strings.HasPrefix(f.Name(), "bar"), Equals, true)
+
+	c.Assert(strings.Index(f.Name(), "bar"), Not(Equals), -1)
```

Comment:
But, this is a specific case that is testing a specific behavior. Then, If you changed some code, the output of the file system should be the same. 

Another good practice, at least for me, is to have assets the most specific possible. In this case we can swallow a lot of bug related to path names doing this.